src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -0,0 +1,104 @@
+package com.microsoft.cse.helium.app.services.keyvault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Value;
+import java.io.IOException;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.MSICredentials;
+import com.microsoft.azure.credentials.AzureCliCredentials;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.CertificateBundle;
+import com.microsoft.azure.keyvault.models.KeyBundle;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+
+@Service
+public class KeyVaultService implements IKeyVaultService
+{
+    private String _keyVaultName = "";
+    private String _environmentFlag = "";
+    private AzureTokenCredentials _credentials;
+    private KeyVaultClient _keyVaultClient;
+
+    private final String _keyVaultNameRegex ="^[a-zA-Z](?!.*--)([a-zA-Z0-9-]*[a-zA-Z0-9])?$";
+    private static final Logger _logger = LoggerFactory.getLogger(KeyVaultService.class);
+
+    public KeyVaultService (@Value("${helium.keyvault.name}") String keyVaultName, @Value("${helium.environment.flag}") String environmentFlag) throws Exception {
+        _keyVaultName = keyVaultName.trim().toUpperCase();
+        _environmentFlag = environmentFlag.trim().toUpperCase();
+
+        // validate params and create credential object
+        if(_environmentFlag.equals("MSI")) {
+            _credentials = new MSICredentials(AzureEnvironment.AZURE);
+        } else if (_environmentFlag.equals("CLI")) {
+            try{
+                _credentials = AzureCliCredentials.create();
+            }
+            catch (IOException ex)
+            {
+                _logger.error(ex.getMessage());
+                throw(ex);
+            }
+        } else {
+            throw new IllegalArgumentException("helium.environment.flag (value='" + _environmentFlag + "') must be 'MSI' or 'CLI'.  Check ${He_EnvironmentFlag} in your environment variables.");
```

Comment:
where is this exception handled? 

One of our goals is to "fail early", so the app should end here (or soon after)

Reply:
It will end, because Spring cannot instantiate the Service in order to inject it.

Reply:
understand, but when spring boot ends with an unhandled exception, it spews hundreds of lines of stack info which makes finding the real issue hard. As long as the Developer Experience is positive ...